### PR TITLE
fix: use tokio Mutex and clear other clippy warnings

### PR DESCRIPTION
**Pull request by jokeyrhyme**
- ran `cargo clippy --fix` and manually applied other fixes for linter warnings
- almost considered this a "chore: ..." PR, but the `tokio::sync::Mutex` adoption seemed like a slight change in behaviour (although nothing observable from outside) :shrug: 
- tested that `cosmic-notifications` and `cosmic-panel` launch as before (can see and interact with panel, and `notify-send ...` results in a notification being displayed)
- tested that `kill`ing the `cosmic-notifications` process causes `cosmic-panel` to be restarted, with both working as expected
- tested that `kill`ing the `cosmic-panel` process causes `cosmic-notifications` to be restarted, with both working as expected

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -149,7 +146,7 @@ async fn start(
 	let notifications_span = info_span!(parent: None, "cosmic-notifications");
 	let panel_span = info_span!(parent: None, "cosmic-panel");
 
-	let mut guard = notif_key.lock().unwrap();
+	let mut guard = notif_key.lock().await;
```
Comment (jokeyrhyme) [inline src/main.rs:149]:
```
warning: this `MutexGuard` is held across an `await` point
...
    = help: consider using an async-aware `Mutex` type or ensuring the `MutexGuard` is dropped before calling await
...
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#await_holding_lock
    = note: this warning originates in the macro `$crate::select` which comes from the expansion of the macro `tokio::select` (in Nightly builds, run with -Z macro-backtrace for more info)

```

File: src/main.rs [hunk 2]
```diff
@@ -343,5 +340,5 @@ async fn start_component(
 				.with_fds(move || vec![fd]),
 		)
 		.await
-		.expect(&format!("failed to start {}", cmd));
+		.unwrap_or_else(|_| panic!("failed to start {}", cmd));
```
Comment (jokeyrhyme) [inline src/main.rs:343]:
```
warning: use of `expect` followed by a function call
...
help: try: `unwrap_or_else(|_| panic!("failed to start {}", cmd))`
...
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#expect_fun_call
    = note: `#[warn(clippy::expect_fun_call)]` on by default
```

File: src/notifications.rs [hunk 3]
```diff
@@ -76,15 +76,15 @@ pub fn notifications_process(
 				.iter_mut()
 				.find(|(k, _v)| k == PANEL_NOTIFICATIONS_FD || k == DAEMON_NOTIFICATIONS_FD)
 			{
-				*v = format!("{}", my_fd.as_raw_fd().to_string());
+				*v = format!("{}", my_fd.as_raw_fd());
```
Comment (jokeyrhyme) [inline src/notifications.rs:79]:
```
warning: `to_string` applied to a type that implements `Display` in `format!` args

   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#to_string_in_format_args
   = note: `#[warn(clippy::to_string_in_format_args)]` on by default
```

File: src/notifications.rs [hunk 4]
```diff
@@ -122,14 +122,14 @@ pub fn notifications_process(
 						error!(?why, "Failed to update fds");
 					}
 
-					let Some(old) = restart_key.lock().unwrap().clone() else {
+					let Some(old) = *restart_key.lock().await else {
```
Comment (jokeyrhyme) [inline src/notifications.rs:125]:
```
warning: using `clone` on type `Option<ProcessKey>` which implements the `Copy` trait
...
help: try dereferencing it: `*restart_key.lock().unwrap()`
...
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#clone_on_copy
    = note: `#[warn(clippy::clone_on_copy)]` on by default
```

File: src/notifications.rs (hunk 3 above)
Comment (jokeyrhyme) [reply src/notifications.rs:79]: Hmmm, for this one, I wonder if keeping `.to_string()` but removing the `format!()` is really how this ought to be done...

File: src/main.rs (hunk 1 above)
Comment (mmstick) [reply src/main.rs:149]: That's a good catch. I haven't seen this lint before.

File: src/notifications.rs (hunk 3 above)
Comment (mmstick) [reply src/notifications.rs:79]: Removing the format would be better.

Comment (jokeyrhyme) [reply src/notifications.rs]: Done

**Comment by nyabinary** [comment]
This should be ready to merge right?

**Review by Drakulix — APPROVED** [review]

**Pull request merged**